.yarn/sdks/typescript/lib/tsserver.js

```diff
@@ -10,11 +10,41 @@ const absPnpApiPath = resolve(__dirname, relPnpApiPath);
 const absRequire = (createRequire || createRequireFromPath)(absPnpApiPath);
 
 const moduleWrapper = tsserver => {
+  const {isAbsolute} = require(`path`);
+  const {resolveVirtual} = require(`pnpapi`);
+
   // VSCode sends the zip paths to TS using the "zip://" prefix, that TS
   // doesn't understand. This layer makes sure to remove the protocol
   // before forwarding it to TS, and to add it back on all returned paths.
 
-  const {isAbsolute} = require(`path`);
+  // We also take the opportunity to turn virtual paths into physical ones;
+  // this makes is much easier to work with workspaces that list peer
+  // dependencies, since otherwise Ctrl+Click would bring us to the virtual
+  // file instances instead of the real ones.
+
+  function toVSCodePath(str) {
+    // We add the `zip:` prefix to both `.zip/` paths and virtual paths
+    if (isAbsolute(str) && !str.match(/^\^zip:/) && (str.match(/\.zip\//) || str.match(/\$\$virtual\//))) {
+      // Absolute VSCode `Uri.fsPath`s need to start with a slash.
+      // VSCode only adds it automatically for supported schemes,
+      // so we have to do it manually for the `zip` scheme.
+      // The path needs to start with a caret otherwise VSCode doesn't handle the protocol
+      // https://github.com/microsoft/vscode/issues/105014#issuecomment-686760910
+      return `${isVSCode ? '^' : ''}zip:${resolveVirtual(str).replace(/^\/?/, `/`)}`;
```

Comment:
It probably works but this references `isVSCode` before it is defined

Reply:
It's fine, it would crash unless `isVSCode` is properly initialized before the function is called ([Temporal Dead Zone](https://wesbos.com/temporal-dead-zone)).